### ISSUE TITLE
Deleted contact still "is in your contacts"

### ISSUE THREAD

**Issue by dscotese** [signalapp/Signal-Android#10092]
- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
If you delete from your contacts a contact with whom you have a Signal conversation and then go to your Signal conversation and click on the contact icon (initials in circle or image) Signal reports that the contact is still "already in your contacts."

### Steps to reproduce
- Find a conversation in Signal with someone for whom your contact info is easily re-entered.
- Find the other party in your system contacts and delete them from your system contacts.
- Find your conversation with them in Signal again.
- Click the circle with their initials or image at the top to view their information.
- Click the message "This person is in your contacts"

**Actual result:** An error notification, "Contact not found."
**Expected result:** The error is fine, ONCE, but if it happens, Signal should update its records so that the message changes from "This person is in your contacts" to "Add to system contacts." 

### Device info

**Device:** Samsung Galaxy 7
**Android version:** 8.0.0
**Signal version:** 4.73.4

### Link to debug log
https://debuglogs.org/8691522e14939d6ab715937384b938ff86c0acd20f5916cd9221a75d6063b117

**Comment by simontb** [comment]
Something that I expect to be related to this bug: One of my friends changed his phone number, so I saved the new one in his contact and deleted the old number. However, the Signal contact stayed. This even led to the strange behavior that in my contacts app both the old and new number were visible, but only the new one editable.

I had to [re-sync my contacts](https://support.signal.org/hc/en-us/articles/360007319011-Manage-Contacts#android_contacts_resync) to get the old contact to disappear.
